src/WebJobs.Script.WebHost/Controllers/KeysController.cs

```diff
@@ -18,11 +21,13 @@ public class KeysController : ApiController
 
         private readonly WebScriptHostManager _scriptHostManager;
         private readonly ISecretManager _secretManager;
+        private readonly TraceWriter _traceWriter;
 
-        public KeysController(WebScriptHostManager scriptHostManager, ISecretManager secretManager)
+        public KeysController(WebScriptHostManager scriptHostManager, ISecretManager secretManager, TraceWriter traceWriter)
         {
             _scriptHostManager = scriptHostManager;
             _secretManager = secretManager;
+            _traceWriter = traceWriter.WithPrefix($"{ScriptConstants.TracePrefixSecretManagement} API");
```

Comment:
Good point! I see that we have overloads of the various tracing methods, but think it would still be helpful to have this method, renamed to `WithSource`, so it automatically sets the source avoiding the need to have that done from every tracing call. Thoughts?